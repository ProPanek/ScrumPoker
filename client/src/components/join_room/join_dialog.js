import React, {Component} from "react"
import Dialog from "@material-ui/core/Dialog"
import DialogTitle from "@material-ui/core/DialogTitle"
import DialogContent from "@material-ui/core/DialogContent"
import DialogContentText from "@material-ui/core/DialogContentText"
import TextField from "@material-ui/core/TextField"
import DialogActions from "@material-ui/core/DialogActions"
import Button from "@material-ui/core/Button"
import {inject, observer} from "mobx-react";
import {withRouter} from "react-router-dom";
import {decorate, observable} from "mobx";
import routes from "routes"

class JoinDialog extends Component {
  constructor(props) {
    super(props)
    this.userName = ""
    this.roomPassword = ""
  }

  componentDidMount() {
    // if (localStorage.getItem("userName") !== null) {
    //   this.userName = JSON.parse(localStorage.getItem("userName"))
    // }
  }

  handleChangUserName = e => {
    this.userName = e.target.value
  };

  handleChangRoomPassword = e => {
    this.roomPassword = e.target.value
  };
  joinRoom = () => {
    const {match: {params}, store: {roomStore: {joinRoom}, socketStore: {openNotification}, userStore: {setUserName}}} = this.props
    if (this.userName && this.roomPassword) {
      setUserName(this.userName)
      joinRoom(this.userName, params.id, this.roomPassword)
    } else {
      openNotification("To join a room you need to fill all inputs", "warning")
    }
  }

  cancelJoinRoom = () => {
    this.props.history.push(routes.root())
  }

  render() {
    const {store: {userStore: {isConnected}}} = this.props
    return (
      <Dialog
        open={!isConnected}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title">Join Room</DialogTitle>
        <DialogContent>
          <DialogContentText>
            Please write your username below if you want to join to the room.
          </DialogContentText>
          <TextField
            autoFocus
            margin="dense"
            id="userName"
            label="User Name"
            type="text"
            fullWidth
            value={this.userName}
            onChange={this.handleChangUserName}
          />
          <TextField
            autoFocus
            margin="dense"
            id="roomPassword"
            label="Room Password"
            type="password"
            fullWidth
            value={this.roomPassword}
            onChange={this.handleChangRoomPassword}
          />
        </DialogContent>
        <DialogActions>
          <Button onClick={this.cancelJoinRoom} color="primary">
            Cancel Join
          </Button>
          <Button onClick={this.joinRoom} color="primary" variant="contained">
            Join Room
          </Button>
        </DialogActions>
      </Dialog>
    )
  }
}

decorate(JoinDialog, {
  userName: observable,
  roomPassword: observable
});

export {JoinDialog}
export default inject("store")(withRouter(observer(JoinDialog)));

