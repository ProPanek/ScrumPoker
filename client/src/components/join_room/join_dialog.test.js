import React from "react";
import {shallow} from "enzyme";
import {JoinDialog} from "./join_dialog";


const noop = () => {}
describe("JoinRoom", () => {

  it("renders without crashing", () => {
    const props = {
      store: {
        userStore: {connected: false}
      }
    }
    const component = shallow(<JoinDialog {...props}/>)
    expect(component).toMatchSnapshot()
  });

});