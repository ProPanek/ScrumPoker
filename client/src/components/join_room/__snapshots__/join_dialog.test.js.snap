// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinRoom renders without crashing 1`] = `
<WithStyles(Dialog)
  aria-labelledby="form-dialog-title"
  open={true}
>
  <WithStyles(DialogTitle)
    id="form-dialog-title"
  >
    Join Room
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <WithStyles(DialogContentText)>
      Please write your username below if you want to join to the room.
    </WithStyles(DialogContentText)>
    <TextField
      autoFocus={true}
      fullWidth={true}
      id="userName"
      label="User Name"
      margin="dense"
      onChange={[Function]}
      required={false}
      select={false}
      type="text"
      value=""
      variant="standard"
    />
    <TextField
      autoFocus={true}
      fullWidth={true}
      id="roomPassword"
      label="Room Password"
      margin="dense"
      onChange={[Function]}
      required={false}
      select={false}
      type="password"
      value=""
      variant="standard"
    />
  </WithStyles(DialogContent)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      color="primary"
      onClick={[Function]}
    >
      Cancel Join
    </WithStyles(Button)>
    <WithStyles(Button)
      color="primary"
      onClick={[Function]}
      variant="contained"
    >
      Join Room
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;
